### PR TITLE
[ISSUE #1653] Refactor Code about client register.

### PR DESCRIPTION
**Pull request by KevinClair**
Feature #1653 
* Use factory to refactor code about client register.

Make sure that:

- [ ] You have read the [contribution guidelines](https://dromara.org/projects/soul/contributor/).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/ShenyuClientRegisterTarsServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * tars service register.
+ *
+ * @author KevinClair
```
Comment (dengliming) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/ShenyuClientRegisterTarsServiceImpl.java:55]: Let's remove `@author` : )

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractShenyuClientRegisterService.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Abstract strategy.
+ *
+ * @author KevinClair
+ **/
+public abstract class AbstractShenyuClientRegisterService implements ShenyuClientRegisterServiceFactory {
```
Comment (dengliming) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractShenyuClientRegisterService.java:44]: What about `AbstractShenyuClientRegisterServiceImpl implements ShenyuClientRegisterService`？

**Review by dengliming — COMMENTED** [review]
Looks great

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/ShenyuClientRegisterTarsServiceImpl.java (hunk 1 above)
Comment (KevinClair) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/ShenyuClientRegisterTarsServiceImpl.java:55]: : )got it.

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractShenyuClientRegisterService.java (hunk 2 above)
Comment (KevinClair) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/AbstractShenyuClientRegisterService.java:44]: OK, I will modify the name.

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/ShenyuClientRegisterSofaServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    @Override
+    public synchronized String register(final MetaDataRegisterDTO dto) {
+        MetaDataDO metaDataDO = metaDataMapper.findByPath(dto.getPath());
+        if (Objects.nonNull(metaDataDO)
+                && (!metaDataDO.getMethodName().equals(dto.getMethodName())
+                || !metaDataDO.getServiceName().equals(dto.getServiceName()))) {
+            return "you path already exist!";
+        }
```
Comment (dengliming) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/ShenyuClientRegisterSofaServiceImpl.java:86]: Can we add this part to `AbstractShenyuClientRegisterService`. Both sofa and tars can use it.

**Review by dengliming — APPROVED** [review]

**Pull request merged**